### PR TITLE
lets add PVC for jenkins #4413

### PR DESCRIPTION
**Pull request by jstrachan**

### CODE REVIEW

**Comment by jstrachan** [comment]
ok to test

**Comment by jstrachan** [comment]
[merge]

File: jenkins/src/main/fabric8/jenkins-deployment.yml [hunk 1]
```diff
@@ -107,6 +107,12 @@ spec:
...
           readOnly: false
+        - mountPath: /var/jenkins_home/jobs
+          name: jenkins-jobs
+          readOnly: false
+        - mountPath: /var/jenkins_home/workspace
+          name: jenkins-workspace
+          readOnly: false
       serviceAccountName: "jenkins"
       volumes:
       - hostPath:
```
Comment (jimmidyson) [inline jenkins/src/main/fabric8/jenkins-deployment.yml]: As discussed on #544, host path volumes should be avoided.

Comment (jstrachan) [inline jenkins/src/main/fabric8/jenkins-deployment.yml]: this is a PVC - its up to gofabric8 or the admin user to set up the PV

Comment (jimmidyson) [inline jenkins/src/main/fabric8/jenkins-deployment.yml]: Doh sorry this is mounting the docker socket - which I thought we no longer needed with @iocanel's latest updates, at least on OpenShift?

Comment (jstrachan) [inline jenkins/src/main/fabric8/jenkins-deployment.yml]: Once we have the S2I binary builds and ImageStreams working with Arquillian, Jenkins Pipelines and multiple environments; yes we'll be able to remove access to the docker socket on OpenShift. We'll probably need 2 apps really; the docker access flavour and the openshift-only-no-docker-socket flavour

**Pull request merged**